### PR TITLE
Added ability to pad images to PrepImageForClipVision Node

### PR DESCRIPTION
**Pull request by Arron17**
Code taken from a node in @laksjdjf implementation. Adds a boolean switch to the "Prepare Image For Clip Vision" node to create padding for the input image.

This adds black bars to the image if it isn't square so the entire contents of the input image are kept and the black bars keep the image at the correct size. I have found this to be helpful with getting the contents of all of a wide image into IPAdapter, however it does have the caveat that at high IPAdapter weights, it can pull the black bars into the image.

### CODE REVIEW

**Comment by Arron17** [comment]
Example of how it can help:
Input Image:
image

Without Padding limited to picking a section of the tall image, in this case top:
image

Resulting Generation:
image


If padding is used like this:
image

Then the resulting image contains the other elements of the image that were missed by being forced to just the top of the image:
image

**Comment by cubiq** [comment]
I'm okay with this, a couple of questions:
- why do you use the output_size parameter?
- have you considered using the average color for the background instead of black? that might help lowering the chance of getting bands

**Comment by Arron17** [comment]
Ah, on Output_size, that's a mistake. It's from an experiment I did with having changeable output sizes to see how IPAdapter reacted, but it didn't make enough of a difference to be useful, so that can be removed.

I'll have a look at using the average colour and update this request.

**Comment by Arron17** [comment]
I've done some testing with using different colours for padding, White, Black and Average. There doesn't seem to be a huge difference between white and black at a high weight of 1.0, but the average seems to cause some issues. All still have black bars.

image
image

image
image

image
image


With a lower weight like 0.8 there is less difference between the images, but they all still have black bars.


For implementing the padding with colours you cannot use torch.nn.functional.pad on the tensor as that only allows grayscale padding. Instead when testing I implemented it with Pillow and PIL.ImageOps.pad.

What do you think would be the best way to do this? I don't think there is any real way to absolutely get rid of black bars at high weights without introducing other problems.

**Comment by cubiq** [comment]
let me run some tests, I'll get back to you

**Comment by cubiq** [comment]
okay made quite a few tests, it seems that at the end the best option is to pad with 0. I was able to remove the bands at 0.6 strength with ipadapter plus, possibly you can do better with accurate negative prompting.

only thing I'd suggest is to move the option in the "crop_position" as last element in the array ("pad"), instead of using a new boolean field. I know it's not technically correct, but it's good enough. And I would not use an external function for it, the code is simple enough to be integrated into the main prepimage node.

I can take care of all those things myself. Thanks for your feedback!
